main.go

```diff
@@ -28,4 +46,157 @@ func main() {
 	}
 	go hcServer.Run()
 
+	// setup graceful crawler shutdown
+	cancelSignal := make(chan os.Signal, 1)
+	signal.Notify(cancelSignal, os.Interrupt)
+
+	// setup ticker for recurring data pulls
+	ticker := time.NewTicker(1 * time.Minute)
+
+	// setup an http client
+	httpClient := http.Client{}
+
+	// create a boolean to track if a crawl has been successful
+	var lastCrawlCall time.Time
+
+	for {
+		select {
+		case _ = <-cancelSignal:
+			l.Info().Msg("shutting down gracefully")
+			os.Exit(0)
+		case <-ticker.C:
+			if lastCrawlCall.IsZero() {
+				goto crawling
+			}
+			if time.Now().Sub(lastCrawlCall) < 6*time.Hour {
+				log.Warn().Msg("already called crawl in the last 6h, skipping run")
+				break
+			}
+		crawling:
+			log.Info().Msg("checking for new data on the webpage")
+			res, err := httpClient.Get(crawlUrl)
+			if err != nil {
+				l.Error().Err(err).Msg("error while requesting measurement page")
+				break
+			}
+			page, err := goquery.NewDocumentFromReader(res.Body)
+			if err != nil {
+				l.Error().Err(err).Msg("unable to parse response into document")
+				break
+			}
+
+			rows := page.Find(`tr[id^="` + tableID + `"]`)
+
+			var stations []types.Station
+			var measurements []types.Measurement
+			var errorOccurred bool
+
+			rows.Each(func(i int, row *goquery.Selection) {
+				dataFields := row.Find("td").Nodes
+				station := types.Station{}
+				err = station.FromDataFields(dataFields)
+				if err != nil {
+					log.Error().Err(err).Msg("unable to create station from page")
+					errorOccurred = true
+					return
+				}
+				stations = append(stations, station)
+				measurement := types.Measurement{}
+				err = measurement.FromDataFields(dataFields)
+				if err != nil {
+					log.Error().Err(err).Msg("unable to create measurement from page")
+					errorOccurred = true
+					return
+				}
+				measurements = append(measurements, measurement)
+			})
```

Comment:
This part may not be an own function since i need access to the `measurements` and `stations` array and the signature of `rows.Each` does not allow returning values. Making the arrays gobally is also not an option since they may not be empty on the next crawl

Reply:
just pass pointers